signing/src/test/java/tech/pegasys/web3signer/signing/secp256k1/aws/AwsKmsClientTest.java

```diff
@@ -0,0 +1,222 @@
+  private static AwsKmsClient awsKmsClient;
+
+  @BeforeAll
+  static void init() {
+    awsKmsUtil =
+        new AwsKmsUtil(
+            AWS_REGION,
+            RW_AWS_ACCESS_KEY_ID,
+            RW_AWS_SECRET_ACCESS_KEY,
+            AWS_SESSION_TOKEN,
+            ENDPOINT_OVERRIDE);
+    testKeyId = awsKmsUtil.createKey(Collections.emptyMap());
+    testWithTagKeyId = awsKmsUtil.createKey(Map.of("name", "tagged"));
+    testWithDisabledKeyId = awsKmsUtil.createKey(Map.of("name", "disabled"));
+    awsKmsUtil.disableKey(testWithDisabledKeyId);
+    testWithNistSecpKeyId = awsKmsUtil.createKey(Map.of("name", "nist"), KeySpec.ECC_NIST_P256);
+
+    final AwsCredentialsBuilder awsCredentialsBuilder = AwsCredentials.builder();
+    awsCredentialsBuilder
+        .withAccessKeyId(AWS_ACCESS_KEY_ID)
+        .withSecretAccessKey(AWS_SECRET_ACCESS_KEY);
+    AWS_SESSION_TOKEN.ifPresent(awsCredentialsBuilder::withSessionToken);
+
+    final AwsCredentialsProvider awsCredentialsProvider =
+        AwsCredentialsProviderFactory.createAwsCredentialsProvider(
+            AwsAuthenticationMode.SPECIFIED, Optional.of(awsCredentialsBuilder.build()));
+
+    final KmsClientBuilder kmsClientBuilder = KmsClient.builder();
+    kmsClientBuilder.credentialsProvider(awsCredentialsProvider).region(Region.of(AWS_REGION));
+    ENDPOINT_OVERRIDE.ifPresent(kmsClientBuilder::endpointOverride);
+
+    awsKmsClient = new AwsKmsClient(kmsClientBuilder.build());
+  }
+
+  @AfterAll
+  static void cleanup() {
+    if (awsKmsUtil == null) {
+      return;
+    }
+    // delete key
+    awsKmsUtil.deleteKey(testKeyId);
+    awsKmsUtil.deleteKey(testWithTagKeyId);
+  }
+
+  @Test
+  void keyPropertiesCanBeMappedUsingCustomMappingFunction() {
+    final MappedResults<String> result =
+        awsKmsClient.mapKeyList(
+            KeyListEntry::keyId, Collections.emptyList(), Collections.emptyList());
+
+    final Optional<String> testKeyEntry =
+        result.getValues().stream().filter(e -> e.equals(testKeyId)).findAny();
+    Assertions.assertThat(testKeyEntry).isPresent();
+    Assertions.assertThat(testKeyEntry.get()).isEqualTo(testKeyId);
+    Assertions.assertThat(result.getErrorCount()).isZero();
+  }
+
+  @Test
+  void mapKeyPropertiesThrowsAwayObjectsWhichFailMapper() {
+    final MappedResults<String> result =
+        awsKmsClient.mapKeyList(
+            kl -> {
+              if (kl.keyId().equals(testKeyId)) {
+                throw new IllegalStateException("Failed mapper");
+              } else {
+                return kl.keyId();
+              }
+            },
+            Collections.emptyList(),
+            Collections.emptyList());
+
+    final Optional<String> testKeyEntry =
+        result.getValues().stream().filter(e -> e.equals(testKeyId)).findAny();
+    Assertions.assertThat(testKeyEntry).isEmpty();
+    Assertions.assertThat(result.getErrorCount()).isOne();
+  }
+
+  @Test
+  void mapKeyPropertiesUsingTagsKey() {
```

Comment:
should these names be updated to `mapKeyList...`?

Reply:
Yes, they should be. Updated them